src/index.js

```diff
@@ -183,20 +183,56 @@ export default function (babel) {
 
 
   // HELPER FUNCTIONS
+  function visitTailExpressions(path, allowLoops, visitor) {
+    path.resync();
 
-  function transformTerminalExpressionsIntoArrPush(path, arrId) {
-    path.resync(); // not sure if this is necessary... c/p from addImplicitReturns
+    let tailPaths = getTailExpressions(path.get("body"), allowLoops);
+    for (let tailPath of tailPaths) {
+      if (tailPath.isExpressionStatement()) {
+        visitor(tailPath);
+      } else if (tailPath.isFunctionDeclaration()) {
+        // Promote function declarations to exprs.
```

Comment:
> I think a FunctionExpression with a name gets pretty printed by Babel into a declaration in the final code.

I don't believe so; Babel will either wrap it in parentheses if it's not already part of some expression.